src/librustdoc/clean/types.rs

```diff
@@ -1966,11 +1967,64 @@ crate struct Static {
 }
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-crate struct Constant {
-    crate type_: Type,
-    crate expr: String,
-    crate value: Option<String>,
-    crate is_literal: bool,
+crate enum Constant {
+    /// Typed constant value.
```

Comment:
What does this mean? Aren't all values typed?

Reply:
I took the documentation from https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Const.html

If you have ideas for improvements, it's very welcome. :laughing:

Reply:
I \*think\* it might be referring to the fact that [`ConstValue`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/interpret/value/enum.ConstValue.html) is basically loosely-structured raw bytes and you need the type to figure out what it actually means.

Reply:
> I took the documentation from https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Const.html

Ah ok, you literally mean `ty::Const`, I thought it had some deeper meaning. And the reason it can't be combined with `Local` is because non-local constants won't have a `BodyId`. Can you add both those things as a comment? And please remove "A typed constant value" - that would be good documentation for `enum Constant`, but it's not helpful for this variant because it's true for *all* the `Constant` variants.